datafusion/physical-plan/src/sorts/sort.rs

```diff
@@ -415,6 +492,8 @@ impl ExternalSorter {
                 (*max_record_batch_size).max(batch.get_sliced_size()?);
         }
 
+        self.reservation.free();
+
```

Comment:
I changed the timing of when the memory reservation is released; from before spilling to after the spill is completed. Do you think this change is safe or should I keep the original order (release before spill)?

Reply:
because `in_progress_file` does not reserve any memory right now it is fine, however if in progress file will add a `BufWriter` and it should also reserve memory by that buffer than this will be a problem and we should have a "transaction" to move reservation between consumer.

but for now it's ok